modules/vmdatasource/src/xmpdatasource.cpp

```diff
@@ -237,9 +341,16 @@ void XMPDataSource::remove(const vector<IdType>& ids)
     try
     {
         metadataSource->remove(ids);
-        xmpFile.PutXMP(*xmp);
-        closeFile();
-        openFile(this->metaFileName, this->openMode);
+
+        //instead of saving to file just put everything to buffer and get back
+        string tempBuffer;
+        XMP_OptionBits options = kXMP_ReadOnlyPacket | kXMP_UseCompactFormat;
+        xmp->SerializeToBuffer(&tempBuffer, options, 0, NULL);
+
+        xmp = make_shared<SXMPMeta>();
+        xmp->ParseFromBuffer(tempBuffer.c_str(), tempBuffer.size(), 0);
+        schemaSource = make_shared<XMPSchemaSource>(xmp);
+        metadataSource = make_shared<XMPMetadataSource>(xmp);
```

Comment:
this is good as a quick workaround, but the best solution is to analyze why re-reading is required here and fix this...

i suggest we leave a comment here, like "// TODO: existing implementation requires metadata re-reading here, so a quick workaround is made below; to be fixed to eliminate re-reading in long term future."